packages/langium/src/references/linker.ts

```diff
@@ -149,43 +154,53 @@ export class DefaultLinker implements Linker {
 
             get ref() {
                 if (isAstNode(this._ref)) {
-                    // Most frequent case: the target is aready resolved.
+                    // Most frequent case: the target is already resolved.
                     return this._ref;
                 } else if (this._ref === undefined) {
                     // The reference has not been linked yet, so do that now.
-                    this._ref = linker.getLinkedNode(node, refId, reference);
-                } else if (isAstNodeDescription(this._ref)) {
+                    const refData = linker.getLinkedNode(node, refId, reference);
+                    this._ref = refData.node ?? refData.error;
+                    this._nodeDescription = refData.descr;
+                } else if (isAstNodeDescription(this._nodeDescription)) {
```

Comment:
This case needs to be checked before the previous one: when `_nodeDescription` is defined, `_ref` is still undefined. Please switch the two `if` blocks.